docs/faq.md

```diff
@@ -0,0 +1,157 @@
+# Feldera Frequently Asked Questions
+
+## What is Feldera
+
+Feldera is an early-stage start-up building a new data processing
+engine for *streaming data*.  In the rest of this document we will
+write "F" for Feldera.
+
+Using indistry jargon, the F engine can be described in several ways:
+
+- a continuous query engine
+- a standing query engine
+- performing incremental view maintenance
+- implementing materialized views
+
+## What is streaming data?
+
+Data that is being continuously modified, usually at a high rate
+(using insertions, deletions, and updates).  F is designed for
+database-like data: numbers, strings, dates, JSON.
+
+## How about streaming video?
+
+F is not designed for media processing.
+
+## What are the "killer apps" for F?
+
+There are several types of problems for which F may be the *only*
+viable solution:
+
+- when data arrives *too fast* to process using other tools.  For
+  example, other tools take 5 minutes to perform a result needed every
+  minute.
+
+- when the analysis need to combine data from *multiple sources* or
+  when the analysis is *complex*: other tools are frequently limited
+  in these dimensions
+
+- when data analysis is *too expensive*.  Sometimes you *can* use a
+  very expensive parallel engine to analyze large data volumes, but
+  the analysis would be much cheaper using F, since it is using many
+  fewer resources.
+
+## How is the F engine programmed?
+
+F is programmed in SQL.  F supports essentially the entire SQL
+language.  There exists a low-level Rust API as well, but F's tooling
+is geared towards the SQL API.
+
+## How is F different from streaming engines?
+
+Existing streaming engines offer a subset of F's capabilities.
+
+Like a database, F supports arbitrary insertions and deletions for
+arbitrary data sources.  Many streaming engines only support
+insertions.
+
+F is programmed using the SQL database language.
```

Comment:
It's worth describing the complexity of SQL we can support that other streaming engines do not. You may want to toss out the subset of SQL logic tests that pass with Feldera.

Reply:
All of them have passed at some point. But these are in the *intersection* of all SQLs, not the union. They are not a good measure of coverage, only of correctness.

Reply:
Ack. Then it's worth calling out the breadth of what we can support and make it clear that this is just plain hard for others to pull off even if they tried, because they don't have dbsp powering them.